modules/dynamic-operator/controllers/kuberlogicservice-env/env.go

```diff
@@ -104,20 +107,50 @@ func (e *EnvironmentManager) SetupEnv(ctx context.Context) error {
 	}
 
 	if e.cfg.SvcOpts.TLSSecretName != "" {
-		if _, err := controllerruntime.CreateOrUpdate(ctx, e.Client, tlsSecret, func() error {
-			srcSecret := &v1.Secret{
+		if e.kls.Spec.UseLetsencrypt {
+			letsencryptIssuer := &certmanagerv1.ClusterIssuer{
```

Comment:
Could you please add tests for the logic

Reply:
Done